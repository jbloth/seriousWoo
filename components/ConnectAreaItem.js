import { breakPoints } from '../styles/theme';

const ConnectAreaItem = ({ children, title }) => {
  return (
    <li className="connectArea__item">
      <p className="connectArea__cta">
        <span>{title}</span> <span className="connectArea__arrow">:</span>
      </p>
      {children}

      <style jsx>{`
        .connectArea__item {
          display: flex;
          align-items: center;
        }

        .connectArea__cta {
          display: inline-block;
          width: 11rem;
          font-weight: bold;
        }

        @media only screen and (max-width: ${breakPoints.bp_md}) {
          .connectArea__item {
            margin: 10px 0 18px 0;
            width: 100%;
            flex-direction: column;
            align-items: flex-start;
          }

          .connectArea__cta {
            width: unset;
          }
        }

        @media only screen and (max-width: ${breakPoints.bp_small}) {
        }
      `}</style>
    </li>
  );
};

export default ConnectAreaItem;
