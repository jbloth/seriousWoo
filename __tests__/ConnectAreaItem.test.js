import ConnectAreaItem from '../components/ConnectAreaItem';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import toJSON from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const itemTitle = 'Title';

describe('<ConnectAreaItem/>', () => {
  it('renders', () => {
    shallow(<ConnectAreaItem title={itemTitle}>Child</ConnectAreaItem>);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<ConnectAreaItem title={itemTitle}>Child</ConnectAreaItem>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('updates when props changes', () => {
    const wrapper = shallow(<ConnectAreaItem title={itemTitle}>Child</ConnectAreaItem>);
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.setProps({ title: 'new title' });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
