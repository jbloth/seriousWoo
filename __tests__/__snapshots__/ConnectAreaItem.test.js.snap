// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConnectAreaItem/> matches snapshot 1`] = `
<li
  className="jsx-2134147660 connectArea__item"
>
  <p
    className="jsx-2134147660 connectArea__cta"
  >
    <span
      className="jsx-2134147660"
    >
      Title
    </span>
     
    <span
      className="jsx-2134147660 connectArea__arrow"
    >
      :
    </span>
  </p>
  Child
  <JSXStyle
    dynamic={
      Array [
        "56.25em",
        "44em",
      ]
    }
    id="1202760200"
  >
    .connectArea__item.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.connectArea__cta.__jsx-style-dynamic-selector{display:inline-block;width:11rem;font-weight:bold;}@media only screen and (max-width:56.25em){.connectArea__item.__jsx-style-dynamic-selector{margin:10px 0 18px 0;width:100%;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:flex-start;-webkit-box-align:flex-start;-ms-flex-align:flex-start;align-items:flex-start;}.connectArea__cta.__jsx-style-dynamic-selector{width:unset;}}@media only screen and (max-width:44em){}
  </JSXStyle>
</li>
`;

exports[`<ConnectAreaItem/> updates when props changes 1`] = `
<li
  className="jsx-2134147660 connectArea__item"
>
  <p
    className="jsx-2134147660 connectArea__cta"
  >
    <span
      className="jsx-2134147660"
    >
      Title
    </span>
     
    <span
      className="jsx-2134147660 connectArea__arrow"
    >
      :
    </span>
  </p>
  Child
  <JSXStyle
    dynamic={
      Array [
        "56.25em",
        "44em",
      ]
    }
    id="1202760200"
  >
    .connectArea__item.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.connectArea__cta.__jsx-style-dynamic-selector{display:inline-block;width:11rem;font-weight:bold;}@media only screen and (max-width:56.25em){.connectArea__item.__jsx-style-dynamic-selector{margin:10px 0 18px 0;width:100%;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:flex-start;-webkit-box-align:flex-start;-ms-flex-align:flex-start;align-items:flex-start;}.connectArea__cta.__jsx-style-dynamic-selector{width:unset;}}@media only screen and (max-width:44em){}
  </JSXStyle>
</li>
`;

exports[`<ConnectAreaItem/> updates when props changes 2`] = `
<li
  className="jsx-2134147660 connectArea__item"
>
  <p
    className="jsx-2134147660 connectArea__cta"
  >
    <span
      className="jsx-2134147660"
    >
      new title
    </span>
     
    <span
      className="jsx-2134147660 connectArea__arrow"
    >
      :
    </span>
  </p>
  Child
  <JSXStyle
    dynamic={
      Array [
        "56.25em",
        "44em",
      ]
    }
    id="1202760200"
  >
    .connectArea__item.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.connectArea__cta.__jsx-style-dynamic-selector{display:inline-block;width:11rem;font-weight:bold;}@media only screen and (max-width:56.25em){.connectArea__item.__jsx-style-dynamic-selector{margin:10px 0 18px 0;width:100%;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:flex-start;-webkit-box-align:flex-start;-ms-flex-align:flex-start;align-items:flex-start;}.connectArea__cta.__jsx-style-dynamic-selector{width:unset;}}@media only screen and (max-width:44em){}
  </JSXStyle>
</li>
`;
